Code/GraphMol/RWMol.h

```diff
@@ -191,22 +191,33 @@ class RDKIT_GRAPHMOL_EXPORT RWMol : public ROMol {
 
   //@}
 
+  //! Sets groups of atoms with relative stereochemistry
+  /*!
+    \param stereo_groups the new set of stereo groups. All will be replaced.
+
+    Stereo groups are also called enhanced stereochemistry in the SDF/Mol3000
+    file format. stereo_groups should be std::move()ed into this function.
+  */
+  void setStereoGroups(std::vector<StereoGroup> stereo_groups) {
```

Comment:
I think we should either use perfect forwarding here and std::forward or use the generic const std::vector api.

While this is (may be) performant for the current use cases, it won't be in other circumstance.

Reply:
I'll admit that the perfect forwarding stuff makes my head hurt, but why not just use a reference?

Reply:
If data is stored internally a reference is actually worse than pass-by-value.

Because the data is stored internally, the structure wants a copy of the data for itself. If we use a reference, then `setStereoGroups()` will always copy the data into the molecule. If we use pass by value, then the data is copied if the calling code is still using the data. If the calling code is not using it, then it gets moved into the molecule, saving an allocation/copy/etc.

There's a nice discussion of this here: https://stackoverflow.com/a/7592741/1309332

I think that this wrapper is exactly what perfect forwarding is for. I'll test it out.

Reply:
I think you are correct.  I also think I was getting confused with:

```
template<class T>
function(T&& arg) {
  T foo = std::forward(arg);
}
```

Which behaves differently with type deduction (i.e. T&&).

I also just noticed that clang-tidy has a pass-by-value option:  https://clang.llvm.org/extra/clang-tidy/checks/modernize-pass-by-value.html

Reply:
I also expect that in most cases the compiler can optimize the calls with const references, that would be easy to investigate.

Interesting (if confusing reading) 
http://notagrammar.blogspot.com/2016/05/c-pass-by-value-passing-fad.html